### PR TITLE
Fix/issue179 formatter props

### PR DESCRIPTION
**Pull request by davecoates**

### CODE REVIEW

**Comment by davecoates** [comment]
Widget docs are incomplete - will do that next

File: doc-site/pages/docs/ui/index.mdx [hunk 1]
```diff
@@ -33,25 +30,158 @@ for [antd](https://ant.design) or you can implement your own.
...
+`getFormatterForField`:
+
+```js
+import { getFormatterForField as defaultGetFormatterForField } from '@prestojs/ui';
+
+function getFormatterForField(field) {
+    if (field instanceof BooleanField) {
+        // You can just return the formatter component directly but it's good practice to return
+        // the component & `field.getFormatterProps` which allows per field customisations.
+        return [BooleanEmojiFormatter, field.getFormatterProps()];
```
Comment (vbudovski-alliance) [inline doc-site/pages/docs/ui/index.mdx:108]: Could probably expand on *how* it allows this customisation/why it exists.

File: doc-site/pages/docs/ui/index.mdx [hunk 2]
```diff
@@ -33,25 +30,158 @@ for [antd](https://ant.design) or you can implement your own.
...
+provided by the existing [BooleanFormatter](doc:BooleanFormatter) component:
+
+```js
+import { getFormatterForField as defaultGetFormatterForField } from '@prestojs/ui';
+
+function getFormatterForField(field) {
+    if (field instanceof BooleanField) {
+        return [
+            BooleanFormatter,
+            { trueLabel: '✅', falseLabel: '❌', blankLabel: '❓', ...field.getFormatterProps() },
```
Comment (vbudovski-alliance) [inline doc-site/pages/docs/ui/index.mdx]: Would it make more sense to destructure first, then override properties?

File: doc-site/pages/examples/ui/FieldFormatter/basic.tsx [hunk 3]
```diff
@@ -7,17 +16,18 @@ import {
...
     // Provide override for CurrencyField
     if (field instanceof CurrencyField) {
         return [
             NumberFormatter,
-            { localeOptions: { style: 'currency', currency: 'USD', currencyDisplay: 'code' } },
+            {
+                localeOptions: { style: 'currency', currency: 'USD', currencyDisplay: 'code' },
+                // It's good practice to always pass this through so any per field customisations
+                // are respected
+                ...field.getFormatterProps(),
```
Comment (vbudovski-alliance) [inline doc-site/pages/examples/ui/FieldFormatter/basic.tsx:29]: As above, should this come first in the object?

**Comment by vbudovski-alliance** [comment]
@davecoates There are a couple of linter warnings too (unused defs).

File: doc-site/pages/docs/ui/index.mdx (hunk 2 above)
Comment (davecoates) [reply doc-site/pages/docs/ui/index.mdx]: No - this allows a specific field to override the general defaults

**Comment by vbudovski-alliance** [comment]
LGTM

**Pull request merged**